4.9.c - DOC-2999: k8s minor version skip (#11226)
spectrocloud/librarium: docs/docs-content/enterprise-version/upgrade/upgrade.md

:::

### Kubernetes Version Constraint

Enterprise Cluster (EC) binary and Palette Management Appliance installations bundle a specific Kubernetes version with
each Palette release. Kubernetes does not support skipping a minor version during a cluster upgrade, so a Palette

> upgrade cannot cross more than one Kubernetes minor version in a single hop. An upgrade that would skip a Kubernetes

Reviewer: ```suggestion
upgrade cannot cross more than one Kubernetes minor version. An upgrade that would skip a Kubernetes
```

> minor version fails mid-run and leaves the management cluster in an unrecoverable state.

Reviewer: ```suggestion
minor version may fail mid-run and leave the management cluster in an unrecoverable state.
```

The following table lists the Kubernetes version bundled with each recent Palette release.

| Palette Release                | Kubernetes Version |
| :----------------------------- | :----------------: |
| 4.7.40, 4.7.43                 |       1.31.8       |
| 4.8.54, 4.8.56, 4.8.58, 4.8.61 |       1.32.9       |
| 4.9.5, 4.9.8, 4.9.14           |      1.33.10       |
| 4.9.23 and later               |       1.34.6       |

Direct upgrades from any `4.8.x` release to `4.9.23` or later are not supported. The `4.8.x` series ships Kubernetes
`1.32.9`, and `4.9.23` and later ship Kubernetes `1.34.6`, which skips `1.33.x`. To reach `4.9.23` or later from

> `4.8.x`, upgrade in two hops.

Reviewer: ```suggestion
`4.8.x`, upgrade in two steps.
```

1. Upgrade to a `4.9.x` release on Kubernetes `1.33.10`. We recommend `4.9.14`.
2. After the cluster returns to a healthy state, upgrade to the target `4.9.23` or later release.

Before you start any Palette upgrade, compare the Kubernetes version of your current release with that of the target

> release. If the delta is two or more minor versions, plan an intermediate hop through a release on the missing minor.

Reviewer: ```suggestion
release. If the delta is two or more minor versions, plan an intermediate upgrade through a release on the missing minor.
```
